packages/x-data-grid/src/hooks/features/rows/useGridRows.ts

```diff
@@ -387,7 +387,7 @@ export const useGridRows = (
           },
         };
       });
-      apiRef.current.publishEvent('rowsSet');
+      apiRef.current.applySorting();
```

Comment:
Hi @MBilalShafi , do you think we should avoid publishing `rowsSet` event here, to avoid performing unnecessary filter calculations each time a row is reordered?